src/modules/Exchange.ts

```diff
@@ -109,6 +115,83 @@ export class Exchange {
     );
   }
 
+  public async proxyDepositWithZeroEx(
+    {
+      humanSellAmount,
+      sellTokenAddress,
+      starkKey,
+      positionId,
+      slippagePercentage = '1.0',
+    }: {
+      humanSellAmount: string,
+      sellTokenAddress: string,
+      starkKey: string,
+      positionId: BigNumberable,
+      slippagePercentage?: string,
+    },
+    options?: SendOptions,
+  ): Promise<TxResult> {
+    const sellAmount: string = humanCollateralAmountToUint256(humanSellAmount);
+    const isProduction: boolean = this.contracts.networkId === 1;
+
+    const zeroExRequest = await getZeroExSwapQuote(
+      {
+        sellAmount,
+        sellTokenAddress,
+        buyTokenAddress: isProduction
+          ? USDC_ADDRES_PRODUCTION
+          : USDC_ADDRES_STAGING,
+        slippagePercentage,
+        isProduction,
+      },
+    );
+
+    const proxyDepositFunctionSignature = 'deposit(address,uint256,address,uint256,uint256,bytes)';
+    return this.contracts.send(
+      this.contracts.proxyDepositContract,
+      this.contracts.proxyDepositContract.methods[proxyDepositFunctionSignature](
+        sellTokenAddress,
+        sellAmount,
+        isProduction
+          ? 'placeholder for zeroExExchange wrapper'
+          : 'placeholder for zeroExExchange wrapper',
+        starkKeyToUint256(starkKey),
+        bignumberableToUint256(positionId),
+        encode(zeroExRequest.to, zeroExRequest.data),
+      ),
+      options,
+    );
+  }
+
+  public async predictedProxyDepositOutput(
+    {
+      humanSellAmount,
+      sellTokenAddress,
+      slippagePercentage = '1.0',
```

Comment:
Should we sanitize these inputs at all? For example, what if someone tries to pass in a negative value or value greater than `100.0` for `slippagePercentage`?

Reply:
I can add sanity checks